src/picosvg/svg_path_iter.py

```diff
@@ -13,16 +13,69 @@
 # limitations under the License.
 
 import re
+from typing import Generator, Tuple
 from picosvg import svg_meta
 
 _CMD_RE = re.compile(f'([{"".join(svg_meta.cmds())}])')
+_SEPARATOR_RE = re.compile("[, ]+")
+_FLOAT_RE = re.compile(
+    r"[-+]?"  # optional sign
+    r"(?:"
+    r"(?:0|[1-9][0-9]*)(?:\.[0-9]+)?"  # int or float
+    r"|"
+    r"(?:\.[0-9]+)"  # float with leading dot (e.g. '.42')
+    r")"
+    r"(?:[eE][-+]?[0-9]+)?"  # optional scientific notiation
+)
+_BOOL_RE = re.compile("^[01]")
+_ARC_ARGUMENT_TYPES = (
+    (float, _FLOAT_RE),  # rx
+    (float, _FLOAT_RE),  # ry
+    (float, _FLOAT_RE),  # x-axis-rotation
+    (int, _BOOL_RE),  # large-arc-flag
+    (int, _BOOL_RE),  # sweep-flag
+    (float, _FLOAT_RE),  # x
+    (float, _FLOAT_RE),  # y
+)
 
 # https://www.w3.org/TR/SVG11/paths.html#PathDataMovetoCommands
 # If a moveto is followed by multiple pairs of coordinates,
 # the subsequent pairs are treated as implicit lineto commands
 _IMPLICIT_REPEAT_CMD = {"m": "l", "M": "L"}
 
 
+def _parse_args(cmd: str, args: str) -> Generator[float, None, None]:
+    raw_args = [s for s in _SEPARATOR_RE.split(args) if s]
+    if not raw_args:
+        return
+
+    # reverse so we can pop()/append() like a stack, quicker than pop(0)/insert(0)
+    raw_args.reverse()
+
+    if cmd.upper() == "A":
+        arg_types = _ARC_ARGUMENT_TYPES
+    else:
+        arg_types = ((float, _FLOAT_RE),)
+    n = len(arg_types)
+
+    i = 0
+    while raw_args:
+        arg = raw_args.pop()
+        # modulo to wrap around
+        converter, regex = arg_types[i % n]
+        m = regex.match(arg)
+        if not m:
+            raise ValueError(f"Invalid argument #{i} for '{cmd}': {arg!r}")
+
+        start, end = m.span()
+        yield converter(arg[start:end])
```

Comment:
Why not capture a group?

Reply:
I need the span().end to so that I can check if there's any remainder, so I might as well use it?

Reply:
also I want the whole FLOAT_RE match, not any particular part of it, so a group doesn't seem very useful here